src/USER-MISC/fix_pafi.cpp

```diff
@@ -0,0 +1,798 @@
+  idregion = NULL; // not used
+  int iarg = 7;
+  while (iarg < narg) {
+    if (strcmp(arg[iarg],"region") == 0) {
+      if (iarg+2 > narg) error->all(FLERR,"Illegal fix hp command");
+      iregion = domain->find_region(arg[iarg+1]);
+      if (iregion == -1)
+        error->all(FLERR,"Region ID for fix hp does not exist");
+      int n = strlen(arg[iarg+1]) + 1;
+      idregion = new char[n];
+      strcpy(idregion,arg[iarg+1]);
+      iarg += 2;
+    } else if (strcmp(arg[iarg],"overdamped") == 0) {
+      od_flag = force->inumeric(FLERR,arg[iarg+1]);
+      iarg += 2;
+    } else if (strcmp(arg[iarg],"com") == 0) {
+      com_flag = force->inumeric(FLERR,arg[iarg+1]);
+      iarg += 2;
+    } else error->all(FLERR,"Illegal fix hp command");
+  }
+  force_flag = 0;
+
+  for(int i = 0; i < 10; i++) {
+    c_v[i] = 0.;
+    c_v_all[i] = 0.;
+  }
+  for(int i=0; i<5; i++) {
+    proj[i] = 0.0;
+    proj_all[i] = 0.0;
+  }
+  for(int i=0; i<4; i++) {
+    results[i] = 0.0;
+    results_all[i] = 0.0;
+  }
+  maxatom = 1;
+  memory->create(h,maxatom,3,"FixPAFI:h");
+
+  // initialize Marsaglia RNG with processor-unique seed
+  random = new RanMars(lmp,seed + comm->me);
+
+  // nve
+  dynamic_group_allow = 1;
+  time_integrate = 1;
+}
+
+/* ---------------------------------------------------------------------- */
+
+FixPAFI::~FixPAFI()
+{
+  if (copymode) return;
+  delete random;
+  delete [] idregion;
+  memory->destroy(h);
+}
+
+/* ---------------------------------------------------------------------- */
+
+int FixPAFI::setmask()
+{
+  int mask = 0;
+  mask |= POST_FORCE;
+  mask |= POST_FORCE_RESPA;
+  mask |= MIN_POST_FORCE;
+  mask |= INITIAL_INTEGRATE;
+  // nve
+  mask |= FINAL_INTEGRATE;
+  mask |= INITIAL_INTEGRATE_RESPA;
+  mask |= FINAL_INTEGRATE_RESPA;
+  return mask;
+}
+
+/* ---------------------------------------------------------------------- */
+
+void FixPAFI::init()
+{
+  // set index and check validity of region
+  // nve
+  dtv = update->dt;
+  dtf = 0.5 * update->dt * force->ftm2v;
+
+  if (iregion >= 0) {
+    iregion = domain->find_region(idregion);
+    if (iregion == -1)
+      error->all(FLERR,"Region ID for fix hp does not exist");
+  }
+
+  if (strstr(update->integrate_style,"respa")) {
+    step_respa = ((Respa *) update->integrate)->step; // nve
+    nlevels_respa = ((Respa *) update->integrate)->nlevels;
+    if (respa_level >= 0) ilevel_respa = MIN(respa_level,nlevels_respa-1);
+    else ilevel_respa = nlevels_respa-1;
+  }
+
+}
+
+void FixPAFI::setup(int vflag)
+{
+  if (strstr(update->integrate_style,"verlet"))
+    post_force(vflag);
+  else
+    for (int ilevel = 0; ilevel < nlevels_respa; ilevel++) {
+      ((Respa *) update->integrate)->copy_flevel_f(ilevel);
+      post_force_respa(vflag,ilevel,0);
+      ((Respa *) update->integrate)->copy_f_flevel(ilevel);
+    }
+}
+
+void FixPAFI::min_setup(int vflag)
+{
+  if( strcmp(update->minimize_style,"fire")!=0 && strcmp(update->minimize_style,"quickmin")!=0 )
```

Comment:
not a big deal right now, but if you use `utils::strmatch(update->minimize_style,"^fire")` it will also match suffix versions, in case one uses accelerated styles.